### ISSUE TITLE
Crash when iCloud sync

### ISSUE THREAD

**Issue by El-Marti** [FrankHausmann/IOSKeePass#30]
When I swipe and click on Sync on a database, the app crashes.
Secondly, I had difficulty to find the way to sync ! I was trying to click on the cloud at the top of the screen but did not find the swipe menu.

I had also a crash after having deleted the database from the recycle bin ! So I had to remove the app and install it again.

**Comment by FrankHausmann** [comment]
mmh, strange i have tested yesterday so many times, everything looks good.

On icloud must be create a Directory structure for a succesfull sync. 
These structure will create on First time you switch icloud sync, and open a KeePass File, make a change on it go back
to the FilesViewsController and on this Back it will create the structure and copy the changed KeePassDB to iCloud.

the Navbar Cloud button in FilesViewController is only an Indicator for the User that iCloud (has no function by touch on it), Nexcloud etc. is actual reachable, if you switch on Cloud Backup and nov Navbar Button is visible, that means you have ether no NetworkConnection to the Cloud provider or the connection parameters like url username, password are wrong (only webdav)

The sync on slide do only get the backup back from iCloud (without metadata) on webdav sync will look for changes of that file an if the file has another modified date it will work, if not nothing happend.

The crash on the recycle is based on a wrong entry inside the tab show a file there is not the iphones local store, sorry i must block the sync method if it is nothing on icloud.

Once a gain the right way in short form:

Swich on Cloud Backup in Settings, close the App and restart again (this must doing only once), select your KeePassDb in fileview after open go to an Entry, set it in Editmode and store it, go back to FileView (so see know a small Progress) 
after finish the progress, you KeepassDB is ont icloud orw webdav.

I hope this helps

**Comment by El-Marti** [comment]
Quick answer, as usual 🙂

Swich on Cloud Backup in Settings ✅ close the App ✅ and restart again ✅ (this must doing only once), select your KeePassDb in fileview ❌ (Crash) after open go to an Entry, set it in Editmode and store it, go back to FileView (so see know a small Progress)
after finish the progress, you KeepassDB is ont icloud orw webdav.

I have tried with my own db and a new one. The app crashes just after opening the database wether it is with TouchId/FaceId or password.

Eloi

Envoyé de mon iPad

**Comment by FrankHausmann** [comment]
It seems to be another issue, if you open an db and the app crashes that has  nothing todo with cloud backup functions.
Other users has Problems with an iPads, what type of iPad you using ?

**Comment by El-Marti** [comment]
I have an iPad Air 4th gen.
I have only the crash after having enabled iCloud sync and restart the app.

I have just tried the same manipulations on my iPhone 13 and I have exactly the same issue.

App crash when I a opening the database AFTER having enabled the iCloud sync in settings. If I disabled the setting and re open the app, everything is ok again.

Eloi

**Comment by FrankHausmann** [comment]
My Develoment environment baesd on IOS 14.8 there are no problems, but on IOS 15 and greater the crash happend
on an simple present(alertController.....  🤮

**Comment by El-Marti** [comment]
Oh OK, I understand ! Good luck with your battle against iOS 15 :-)
